docs/blueprints/referral-manager-v1.md

```diff
@@ -0,0 +1,229 @@
+# Referral Manager V1: Cross-Satellite Referral Link
+
+## Abstract
+
+This document describes how to handle user referrals across satellites.
+
+## Background
+
+Previous referral program only works per satellite. This means a user on us-east-1 satellite can only refer people to join us-east-1.
+
+We want referrals to be valid on all Tardigrade satellites. This will allow the user to select the satellite that is best for them.
+
+### Non-Goals
+A registration page where a user can select their preferred Satellite during registration.
+
+## Design
+
+### Definitions
+
+Referral Manager
+: A stand-alone process running outside of satellites. It should only talk to tardigrade branded satellites.
+
+Referral Manager CLI
+: A CLI that will allow operators to invoke referral link generation.
+
+Referral Link
+: A one-time-use URL that contains a unique invitation token.
+
+Invitation Token
+: A random number generated by Referral Manager for each user in the referral program.
```

Comment:
So each user gets one, and only one, invitation token? Do we not want to allow people to refer more than once?